### PR TITLE
Use collection expressions for builder patterns.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@akhera99 ptal

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/DynamicTests.cs [hunk 1]
```diff
@@ -1492,11 +1492,7 @@ private static void VerifyCustomTypeInfo(Guid customTypeInfoId, ReadOnlyCollecti
             else
             {
                 Assert.Equal(CustomTypeInfo.PayloadTypeId, customTypeInfoId);
-                // Include leading count byte.
-                var builder = ArrayBuilder<byte>.GetInstance();
-                builder.Add((byte)expectedBytes.Length);
-                builder.AddRange(expectedBytes);
-                expectedBytes = builder.ToArrayAndFree();
+                expectedBytes = [(byte)expectedBytes.Length, .. expectedBytes];
```
Comment (CyrusNajmabadi) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/DynamicTests.cs:1495]:
```suggestion
                // Include leading count byte.
                expectedBytes = [(byte)expectedBytes.Length, .. expectedBytes];
```

File: src/ExpressionEvaluator/Core/Test/ResultProvider/ReflectionUtilities.cs [hunk 2]
```diff
@@ -41,19 +41,21 @@ internal static AssemblyLoadContext LoadAssemblies(params Assembly[] assemblies)
 
         internal static Assembly[] GetMscorlib(params Assembly[] additionalAssemblies)
         {
-            var builder = ArrayBuilder<Assembly>.GetInstance();
-            builder.Add(typeof(object).Assembly); // mscorlib.dll
-            builder.AddRange(additionalAssemblies);
-            return builder.ToArrayAndFree();
+            return
+            [
```
Comment (ToddGrun) [inline src/ExpressionEvaluator/Core/Test/ResultProvider/ReflectionUtilities.cs:45]: >[ [](http://example.com/codeflow?start=12&length=1)

Out of curiosity, what does this now generate under the hood?

Comment (ToddGrun) [reply src/ExpressionEvaluator/Core/Test/ResultProvider/ReflectionUtilities.cs:45]: I guess I'm more curious about the change in InstallPackageParentCodeAction.cs as it creates an ImmutableArray from this.

Comment (CyrusNajmabadi) [reply src/ExpressionEvaluator/Core/Test/ResultProvider/ReflectionUtilities.cs:45]: The codegen is great:

image

https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBAGABAlAOgCUBTAMwBtSBjAFygHsYBuTHfIgYUcuvqYwAzoQCSYMAFc6EYNTYZ2AZnxJcXXAG9MuXblh1SAJxgRKnAGy5xUmXNIBBI0YgBPADwOhQ0mDmuAPlwAcVI6AFkhGkYjSihgAAoABwgXMCFcLx8/SlcAbQBdXAgAExKoBmYzLN85KFIhAEodPW0MPQ78AHZcPJbOgbpXJNJGcgTGYAArWjpGwhqc1wAaXAB6Ndx06Nj4whK+foHjwkJisorBau9auIblo46ChQ6AX0xXoA===

Comment (ToddGrun) [reply src/ExpressionEvaluator/Core/Test/ResultProvider/ReflectionUtilities.cs:45]: That is beautiful. Is  ImmutableCollectionsMarshal net8 only?

Comment (CyrusNajmabadi) [reply src/ExpressionEvaluator/Core/Test/ResultProvider/ReflectionUtilities.cs:45]: I don't believe so. You just get it if you have an updated/modern System.Collections.Immutable dll.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**